dao/src/main/java/org/thingsboard/server/dao/alarm/BaseAlarmService.java

```diff
@@ -411,6 +441,17 @@ public long countAlarmsByQuery(TenantId tenantId, CustomerId customerId, AlarmCo
         return alarmDao.countAlarmsByQuery(tenantId, customerId, query);
     }
 
+    @Override
+    public PageData<EntitySubtype> findAlarmTypesByTenantId(TenantId tenantId, PageLink pageLink) {
+        log.trace("Executing findAlarmTypesByTenantId, tenantId [{}]", tenantId);
+        validateId(tenantId, INCORRECT_TENANT_ID + tenantId);
+        if (DEFAULT_ALARM_TYPES_PAGE_LINK.equals(pageLink)) {
```

Comment:
The quals in PagaLink is ok. But I need clarification why you are not always operating with the cache. If you find something new - you can add it to the cache.

Reply:
This implementation is related to our architecture which we have discussed with You and @ashvayka.